### PR TITLE
swarm/network: hive SuggestPeer unit tests

### PR DESCRIPTION
**Pull request by acud**

### CODE REVIEW

File: swarm/network/hive.go [hunk 1]
```diff
@@ -55,8 +56,7 @@ func NewHiveParams() *HiveParams {
 		MaxPeersPerRequest:    5,
 		KeepAliveInterval:     500 * time.Millisecond,
 		RetryInterval:         4200000000, // 4.2 sec
-		//RetryExponent:         2,
-		RetryExponent: 3,
+		RetryExponent:         3,
```
Comment (zelig) [inline swarm/network/hive.go:59]: why 3?

File: swarm/network/hive.go [hunk 2]
```diff
@@ -333,63 +359,67 @@ func (h *Hive) getPotentialNeighbours(depth int) []*Peer {
...
 	// find the bin shallower than depth that has the LEAST peers
 	// TODO refactor in order to avoid accessing hidden member of kademlia
-	h.Kademlia.conns.EachBin(nil, Pof, offset, func(po int, size int, f func(func(pot.Val, int) bool) bool) bool {
+	h.Kademlia.conns.EachBin(h.Kademlia.base, Pof, offset, func(po int, size int, f func(func(pot.Val, int) bool) bool) bool {
 		seqPo++
-
 		// stop if we reach depth, because peers from that point and deeper will be treated differently
-		if depth >= po {
-			//TODO how to handle simbin -1 here
+		if po <= depth {
```
Comment (zelig) [inline swarm/network/hive.go]: comparison is the wrong way round here

File: swarm/network/hive.go [hunk 3]
```diff
@@ -487,13 +517,14 @@ func (h *Hive) getRetriesFromDuration(timeAgo time.Duration) int {
 }
 
 func (h *Hive) isTimeForRetry(d *Peer) bool {
+	//	debug.PrintStack()
```
Comment (zelig) [inline swarm/network/hive.go]: this comment is for `getRetriesFromDuration` ON LINE 511
what is this?

```
 div += (150000 - rand.Int63n(300000)) * div / 1000000
```

File: swarm/network/hive.go (hunk 1 above)
Comment (acud) [reply swarm/network/hive.go:59]: i just removed the commented one. was it 2 in the first place?

File: swarm/network/hive.go (hunk 3 above)
Comment (acud) [reply swarm/network/hive.go]: not sure i understand the question. anyway removed by now

File: swarm/network/hive.go [hunk 4]
```diff
@@ -379,7 +391,7 @@ func (h *Hive) getPotentialBinPeers(offset int, depth int) ([]*Peer, int) {
 	h.Kademlia.conns.EachBin(h.Kademlia.base, Pof, offset, func(po int, size int, f func(func(pot.Val, int) bool) bool) bool {
 		seqPo++
 		// stop if we reach depth, because peers from that point and deeper will be treated differently
-		if po <= depth {
+		if depth >= po {
```
Comment (zelig) [inline swarm/network/hive.go:391]: lol. still

File: swarm/network/hive.go [hunk 5]
```diff
@@ -330,6 +330,18 @@ func (h *Hive) getPotentialPeers() []*Peer {
 	return callablePeers
 }
 
+func insertSort(arr []int, comp func(int, int) bool) {
```
Comment (zelig) [inline swarm/network/hive.go]: ?

**Comment by nolash** [comment]
Discontinued due to https://github.com/ethereum/go-ethereum/pull/18404

**Pull request closed without merging**